tests/cephfs/client_authorize.py

```diff
@@ -0,0 +1,435 @@
+            long_running=True,
+        )
+        out, rc = client[0].exec_command(
+            sudo=True,
+            cmd=f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation read --threads 10 --file-size 4 "
+            f"--files 1000 --files-per-dir 10 --dirs-per-dir 2 --top {kernel_mount_dir}",
+            long_running=True,
+        )
+        out, rc = client[0].exec_command(
+            sudo=True,
+            cmd=f"dd if=/dev/zero of={fuse_mount_dir}/file bs=10M count=10",
+            long_running=True,
+        )
+        out, rc = client[0].exec_command(
+            sudo=True,
+            cmd=f"dd if={fuse_mount_dir}/file of={fuse_mount_dir}/dir_1/file2 bs=10M count=10",
+            long_running=True,
+        )
+        log.info(f"Permissions set for client {client_name} is working")
+        # Create client with read permission on "/" directory & read-write permission on "dir1" directory
+        client_name = "Client_2"
+        kernel_mount_dir = "/mnt/" + "".join(
+            secrets.choice(string.ascii_uppercase + string.digits) for i in range(5)
+        )
+        fuse_mount_dir = "/mnt/" + "".join(
+            secrets.choice(string.ascii_lowercase + string.digits) for i in range(5)
+        )
+        mount_points.extend([kernel_mount_dir, fuse_mount_dir])
+        log.info(
+            f"Testing {client_name} with read permission on root & read-write permission on /dir_1"
+        )
+        fs_util.fs_client_authorize(
+            client[0], fs_name, client_name, "/", "r", extra_params=" /dir_1 rw"
+        )
+        # Mount cephfs on kernel & fuse client
+        fs_util.kernel_mount(
+            client, kernel_mount_dir, mon_node_ip, new_client_hostname=client_name
+        )
+        fs_util.fuse_mount(client, fuse_mount_dir, new_client_hostname=client_name)
+        # Verify write operation on "/" directory fails
+        try:
+            client[0].exec_command(
+                sudo=True,
+                cmd=f"dd if=/dev/zero of={kernel_mount_dir}/file bs=10M count=10",
+            )
+        except CommandFailed as e:
+            log.info(e)
+            log.info(
+                f"Permissions set for client {client_name} is working for kernel mount"
+            )
+        else:
+            log.error(
+                f"Permissions set for client {client_name} is not working for kernel mount"
+            )
+            return 1
+        try:
+            client[0].exec_command(
+                sudo=True,
+                cmd=f"dd if=/dev/zero of={fuse_mount_dir}/file bs=10M count=10",
+            )
+        except CommandFailed as e:
+            log.info(e)
+            log.info(
+                f"Permissions set for client {client_name} is working for fuse mount"
+            )
+        else:
+            log.error(
+                f"Permissions set for client {client_name} is not working for fuse mount"
+            )
+            return 1
```

Comment:
The code is similar across these lines. Would it be possible to create a method that would execute the flow?

Reply:
Sure